lib/suma/mobility.rb

```diff
@@ -20,4 +20,32 @@ def self.int2coord(i)
     raise OutOfBounds, "#{i} must be between -1.8b and 1.8b" unless INTCOORD_RANGE.cover?(i)
     return i * INT2COORD_FACTOR
   end
+
+  # *map_vehicles* is hash of vehicles types e.g. escooters or ebikes containing an array of vehicle hashes
+  # which include properties that will be used in the front-end e.g. coordinates.
+  # This function calculates and sets an offset integer for each disambiguated vehicle coordinates to use that
+  # new position in the front-end. The vehicles new position will spread in a circular manner from that center location.
+  # This is done to avoid disambiguated vehicles from appearing at the same position causing visiblity issues.
+  # Example of two vehicles at location [450000060, -1220000060]:
+  # o: [-10, 0] and [10, 0] => [45000050, -1220000000] and [45000070, -1220000000]
+  def self.offset_disambiguated_vehicles(map_vehicles)
+    map_vehicles.each_value do |vehicles|
+      vehicles.group_by { |v| v[:c] }.each do |_, shared_loc_vehicles|
+        next if shared_loc_vehicles.count <= 1
+        # measured in pixels
```

Comment:
Pixels are screen-dependent. There is no way this could represent the size in pixels.

I think you have the algorithm mixed up here.

You want to:
1- Split the circumference into pieces. There is actually a hidden '1' here as the radius: you are doing `2 * pi / number_of_segments` but what's going on is `1 * 2* pi / number_of_segments`. (a 1-radius circle is sometimes called a 'unit' circle)
2- Convert that distance into an angle. This is what the cos/sin are doing.
3- Give that angle a "magnitude"- that is, establish a point along the angle's vector.
4- We actually don't have consistent units here. Because latitude and longitude is not a square grid, we cannot have a circle with a consistent radius- a radius of "1" will be oval when it's projected onto such a grid. If we had a really big magnitude, and a lot of scooters, we'd notice this (and it gets more severe as you move towards the poles, where latitude stays constant but longitude gets tiny).
5- That said, lat/lng are measured in degrees. There are 360 degrees of each of lat and lng. The values we're dealing with are like `450000000` which is actually `45.0` degrees latitude. So a diameter of "40" here is actually a fractional degree of `0.0000040`. Using the [haversine formula](https://stackoverflow.com/a/11172685), `measure(0, 0, 0.0000040, 0)`, we find this is about 0.44m or 1.5 feet (so two scooters would be offset about 3 feet/1m from each other.

Let me know if this is too in the weeds and I can refactor it a bit to clean it up.